nativeunwind/elfunwindinfo/elfgopclntab.go

```diff
@@ -343,113 +304,324 @@ func (ee *elfExtractor) parseGoPclntab() error {
 		if err != nil {
 			// It seems the Go binary was stripped. So we use the heuristic approach
 			// to get the stack deltas.
-			if data, err = SearchGoPclntab(ef); err != nil {
-				return fmt.Errorf("failed to search .gopclntab: %v", err)
+			if data, err = searchGoPclntab(ef); err != nil {
+				return nil, fmt.Errorf("failed to search .gopclntab: %v", err)
 			}
 		} else {
 			start, err := symtab.LookupSymbolAddress("runtime.pclntab")
 			if err != nil {
-				return fmt.Errorf("failed to load .gopclntab via symbols: %v", err)
+				return nil, fmt.Errorf("failed to load .gopclntab via symbols: %v", err)
 			}
 			end, err := symtab.LookupSymbolAddress("runtime.epclntab")
 			if err != nil {
-				return fmt.Errorf("failed to load .gopclntab via symbols: %v", err)
+				return nil, fmt.Errorf("failed to load .gopclntab via symbols: %v", err)
 			}
 			if start >= end {
-				return fmt.Errorf("invalid .gopclntab symbols: %v-%v", start, end)
+				return nil, fmt.Errorf("invalid .gopclntab symbols: %v-%v", start, end)
 			}
 			data = make([]byte, end-start)
 			if _, err := ef.ReadVirtualMemory(data, int64(start)); err != nil {
-				return fmt.Errorf("failed to load .gopclntab via symbols: %v", err)
+				return nil, fmt.Errorf("failed to load .gopclntab via symbols: %v", err)
 			}
 		}
 	}
+	return data, nil
+}
+
+// Gopclntab is the API for extracting data from .gopclntab
+type Gopclntab struct {
+	data      []byte
+	textStart uintptr
+	numFuncs  int
+
+	version     uint8
+	quantum     uint8
+	ptrSize     uint8
+	funSize     uint8
+	funcMapSize uint8
+
+	functab, funcdata, funcnametab, filetab, pctab, cutab []byte
```

Comment:
Since we're calling `unsafe.String` and returning strings from `funcnametab` and `filetab` maybe we can add a comment here mentioning that these should not be modified.